src/libraries/System.IO.Compression/tests/ZipArchive/zip_UpdateTests.cs

```diff
@@ -337,5 +338,235 @@ public void UpdateUncompressedArchive()
+            {
+                yield return new object[] { string.Empty, encoding };
+                yield return new object[] { "1", encoding };
+                yield return new object[] { new string('x', ushort.MaxValue), encoding };
+                yield return new object[] { new string('y', ushort.MaxValue + 1), encoding };  // Should get truncated
+            }
+
+            foreach (Encoding encoding in new[] { Encoding.Latin1,Encoding.UTF8 }) // Exclude ASCII
+            {
+               yield return new object[] { "ü", encoding };
+               yield return new object[] { new string('ñ', ushort.MaxValue), encoding }; // Should get truncated
+               yield return new object[] { new string('ö', ushort.MaxValue + 1), encoding }; // Should get truncated
+            }
+        }
+
+        // For the specified text, returns 3 strings:
+        // - The specified text, but encoded in the specified encoding, and truncated to the specified max byte length (adjusted to encoding character size).
+        // - The specified text with an extra prefix, no encoding or length modifications.
+        // - The specified text with an extra prefix, but encoded and truncated.
+        private static (string, string, string) GetTestStringsForEncoding(string text, int maxByteLength, Encoding encoding)
+        {
+            string expectedText = GetEncodedStringOfExpectedByteLength(encoding, maxByteLength, text);
+            string updatedText = "Updated text: " + expectedText;
+            string expectedUpdatedText = GetEncodedStringOfExpectedByteLength(encoding, maxByteLength, updatedText);
+
+            return (expectedText, updatedText, expectedUpdatedText);
+        }
+
+        // Gets the specified string as a non-null string, in the specified encoding,
+        // and no longer than the specified byte length, adjusted for the encoding.
+        // If no encoding is specified, UTF8 is the fallback, just like ZipArchive does.
+        private static string GetEncodedStringOfExpectedByteLength(Encoding? encoding, int maxBytes, string? comment)
+        {
+            string nonNullComment = comment ?? string.Empty;
+
+            if (nonNullComment.Length > 0)
+            {
+                encoding ??= GetEncoding(comment);
+                byte[] bytes = encoding.GetBytes(nonNullComment);
+
+                if (bytes.Length > maxBytes)
+                {
+                    int bytesPerChar = encoding.GetMaxByteCount(1);
+
+                    int encodedCharsThatFit = maxBytes / bytesPerChar;
+                    int totalBytesToTruncate = encodedCharsThatFit * bytesPerChar;
+
+                    bytes = bytes[0..totalBytesToTruncate];
+                }
+
+                return encoding.GetString(bytes);
+            }
+
+            return nonNullComment;
+        }
+
+        private static Encoding GetEncoding(string text)
```

Comment:
IMO by using exactly the same code as in the implementation, we are testing the current implementation, not the behavior:

https://teamgaslight.com/blog/testing-behavior-vs-testing-implementation
https://testing.googleblog.com/2013/08/testing-on-toilet-test-behavior-not.html

Reply:
How do you suggest I generate the string with the expected encoding, for the test?

Reply:
I would just use a hardcoded string value that meets given requirements.

Reply:
I changed the test logic so it didn't exactly match the src logic.
I need to also tests strings that have a length of `ushort.MaxValue` so creating a harcoded byte array would be too much.